src/server/pps/server/worker_rc.go

```diff
@@ -490,15 +506,8 @@ func (a *apiServer) createWorkerRc(options *workerOptions) error {
 			},
 			Spec: v1.ServiceSpec{
 				Selector: options.labels,
-				Type:     v1.ServiceTypeNodePort,
-				Ports: []v1.ServicePort{
-					{
-						Port:       options.service.ExternalPort,
-						TargetPort: intstr.FromInt(int(options.service.InternalPort)),
-						Name:       "user-port",
-						NodePort:   options.service.ExternalPort,
-					},
-				},
+				Type:     serviceType,
```

Comment:
Small tweak: I think you can just do `Type: options.service.Type` and save the conditional above, with the exception of the part where you set the `NodePort`. `v1.ServiceType` is actually just a string under the hood so you can assign to it directly.